src/wasm-interpreter.h

```diff
@@ -1388,13 +1388,13 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     NOTE_ENTER("BrOnCast");
     WASM_UNREACHABLE("TODO (gc): br_on_cast");
   }
-  Flow visitRttCanon(RttCanon* curr) {
-    NOTE_ENTER("RttCanon");
-    WASM_UNREACHABLE("TODO (gc): rtt.canon");
-  }
+  Flow visitRttCanon(RttCanon* curr) { return Literal(curr->type); }
   Flow visitRttSub(RttSub* curr) {
-    NOTE_ENTER("RttSub");
-    WASM_UNREACHABLE("TODO (gc): rtt.sub");
+    Flow parent = this->visit(curr->parent);
+    if (parent.breaking()) {
+      return parent;
+    }
+    return Literal(curr->type);
```

Comment:
From the explainer:

```
In order to cast down the type of a struct or array, the aggregate itself must be equipped with a suitable RTT. Attaching runtime type information to aggregates happens at allocation time but is optional. If no RTT is attached then their runtime type is treated as if it was anyref and a down cast to a more specific type will fail.
```

IIUC, this means that `Struct` and `Array` need to own optional Rtts and that Rtts actually need to refer to two heap types: the supertype and subtype of the subtyping relation they witness. The implementation of casts will need to trap if the provided Rtt does not witness the correct subtying relation.

Although this seems to work differently than in the V8 prototype, which uses a depth instead???

Reply:
Looks like this confusion is because I was reading Overview.md when I should have been reading MVP.md. I filed https://github.com/WebAssembly/gc/issues/167 to fix this.